static/js/popover_menus.js

```diff
@@ -87,4 +90,48 @@ export function initialize() {
             compose_mobile_button_popover_displayed = false;
         },
     });
+
+    // compose formatting buttons popover
+    $("body").on("select", ".new_message_textarea, .message_edit_content ", (e) => {
+        hideAll();
+        const textarea = $(e.currentTarget);
+        const caret_offset = position(e.currentTarget);
+        const range = textarea.range();
+        if (!range || !range.length) {
+            // check if user selected a range of text.
+            return;
+        }
+        tippy(e.target, {
+            ...default_popover_props,
+            content: render_formatting_buttons_popover(),
+            placement: "top-start",
+            trigger: "manual",
+            // Shifting 3px to left places `B` above selection.
+            // Extra 5px from top to get popover closer to text.
+            offset: [caret_offset.left - 3, -1 * caret_offset.top + 5],
+            arrow: false,
+            showOnCreate: true,
+            onHidden(instance) {
+                instance.destroy();
+            },
+            onShow(instance) {
+                const $popper = $(instance.popper);
+                // Remove extra padding around the buttons so that
+                // user doesn't click on it by mistake.
+                $popper.find(".tippy-content").css("padding", 0);
+                $popper.on("click", ".formatting_button", (e) => {
+                    const format_type = $(e.currentTarget).find("[data-format]").data("format");
```

Comment:
Maybe `data-format-type` would be a bit cleaner name for the HTML element containing the `format_type`?